pkg/kubernetes/selinux.go

```diff
@@ -0,0 +1,82 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"io/fs"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/suse-edge/edge-image-builder/pkg/http"
+	"github.com/suse-edge/edge-image-builder/pkg/image"
+	"go.uber.org/zap"
+)
+
+const (
+	rke2SELinuxPolicyReleaseURL = "https://github.com/rancher/rke2-selinux/releases/download/%s/%s"
+	k3sSELinuxPolicyReleaseURL  = "https://github.com/k3s-io/k3s-selinux/releases/download/%s/%s"
+
+	rancherSigningKeyURL = "https://rpm.rancher.io/public.key"
+)
+
+type selinuxPolicy struct {
+	downloadURL string
+	rpmName     string
+}
+
+func DownloadSELinuxRPMs(kubernetes *image.Kubernetes, rpmDir, gpgKeysDir string) error {
+	policy := newSELinuxPolicy(kubernetes.Version)
+	policyPath := filepath.Join(rpmDir, policy.rpmName)
+
+	if _, err := os.Stat(policyPath); err != nil {
+		if !errors.Is(err, fs.ErrNotExist) {
+			zap.S().Errorf("Looking for SELinux RPM '%s' failed unexpectedly: %s", policyPath, err)
+		}
+
+		if err = http.DownloadFile(context.Background(), policy.downloadURL, policyPath, nil); err != nil {
+			return fmt.Errorf("downloading selinux policy: %w", err)
+		}
+	}
+
+	signingKeyPath := filepath.Join(gpgKeysDir, "rancher-public.key")
+	if err := http.DownloadFile(context.Background(), rancherSigningKeyURL, signingKeyPath, nil); err != nil {
+		return fmt.Errorf("downloading rancher signing key: %w", err)
+	}
+
+	return nil
+}
+
+func newSELinuxPolicy(kubernetesVersion string) selinuxPolicy {
+	const (
```

Comment:
Nit: I understand defining them in here from a scope perspective, but it feels weird to have the download locations and this version information in two different places instead of right next to each other for a simpler overall picture.

Reply:
I went back and forth with this one and liked that I can simplify the constant names if they were function scoped instead of being defined on package level i.e. `rke2PolicyVersion` instead of `rke2SELinuxRPMPolicyVersion`.

Reply:
I'll move the policy URLs to the function as well.